### PR TITLE
Use FitsIn8 when setting GT_INT_CNS costs

### PR DESCRIPTION
**Pull request by briansull**
When classifying the size of integer constants use new method FitsIn8 and existing method FitsInI32

### CODE REVIEW

**Comment by briansull** [comment]
@dotnet/jit-contrib PTAL

File: src/jit/gentree.h [hunk 1]
```diff
@@ -2530,6 +2530,17 @@ struct GenTreeIntConCommon : public GenTree
...
     }
 
+    bool FitsInI8()
+    {
+        return FitsInI8(IconValue());
+    }
+
+    static bool FitsInI8(ssize_t val)
+    {
+
```
Comment (mikedn) [inline src/jit/gentree.h]: Useless empty line?

File: src/jit/gentree.h [hunk 2]
```diff
@@ -2530,6 +2530,17 @@ struct GenTreeIntConCommon : public GenTree
...
+        return (char)val == val;
```
Comment (mikedn) [inline src/jit/gentree.h]: `signed char` or `int8_t` to make it more obvious?

File: src/jit/gentree.h [hunk 3]
```diff
@@ -2039,7 +2039,7 @@ struct GenTree
 
     inline bool IsIntegralConst() const;
 
-    inline bool IsIntCnsFitsInI32();
+    inline bool IsIntCnsFitsInI32(); // Constant fits in INT32
```
Comment (CarolEidt) [inline src/jit/gentree.h]: Minor: I don't think this comment adds a lot, but if you add it here, you should probably have a similar comment on the I8 variants below.

Comment (briansull) [reply src/jit/gentree.h]: OK

File: src/jit/gentree.h (hunk 2 above)
Comment (briansull) [reply src/jit/gentree.h]: OK

Comment (briansull) [reply src/jit/gentree.h]: OK

**Pull request merged**